### ISSUE TITLE
Two tests failures with MariaDB-10.2.6

### ISSUE THREAD

**Issue by yahonda** [rails/rails#29698]
### Steps to reproduce
These two tests get failures with  MariaDB-10.2.6

```ruby
$ ARCONN=mysql2 bin/test test/cases/defaults_test.rb:116
$ ARCONN=mysql2 bin/test test/cases/adapters/mysql2/virtual_column_test.rb:55
```

### Expected behavior
Both test should finish without failures.

### Actual behavior
```ruby
$ ARCONN=mysql2 bin/test test/cases/defaults_test.rb:116
Using mysql2
Run options: --seed 40252

# Running:

F

Failure:
MysqlDefaultExpressionTest#test_schema_dump_timestamp_includes_default_expression [/home/ubuntu/rails/activerecord/test/cases/defaults_test.rb:116]:
Expected /t\.timestamp\s+"modified_timestamp",\s+default: -> { "CURRENT_TIMESTAMP" }/ to match "# This file is auto-generated from the current state of the database. Instead\n# of editing this file, please use the migrations feature of Active Record to\n# incrementally modify your database, and then regenerate this schema definition.\n#\n# Note that this schema.rb definition is the authoritative source for your\n# database schema. If you need to create the application database on another\n# system, you should be using db:schema:load, not running all the migrations\n# from scratch. The latter is a flawed and unsustainable approach (the more migrations\n# you'll amass, the slower it'll run and the greater likelihood for issues).\n#\n# It's strongly recommended that you check this file into your version control system.\n\nActiveRecord::Schema.define(version: 0) do\n\n  create_table \"timestamp_defaults\", force: :cascade, options: \"ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci\" do |t|\n    t.timestamp \"nullable_timestamp\"\n    t.timestamp \"modified_timestamp\", null: false\n  end\n\nend\n".


bin/test test/cases/defaults_test.rb:114



Finished in 0.419630s, 2.3830 runs/s, 4.7661 assertions/s.

1 runs, 2 assertions, 1 failures, 0 errors, 0 skips
```

```
$ ARCONN=mysql2 bin/test test/cases/adapters/mysql2/virtual_column_test.rb:55
Using mysql2
Run options: --seed 35427

# Running:

F

Failure:
Mysql2VirtualColumnTest#test_schema_dumping [/home/ubuntu/rails/activerecord/test/cases/adapters/mysql2/virtual_column_test.rb:55]:
Expected /t\.virtual\s+"upper_name",\s+type: :string,\s+as: "ucase\(`name`\)"$/i to match "# This file is auto-generated from the current state of the database. Instead\n# of editing this file, please use the migrations feature of Active Record to\n# incrementally modify your database, and then regenerate this schema definition.\n#\n# Note that this schema.rb definition is the authoritative source for your\n# database schema. If you need to create the application database on another\n# system, you should be using db:schema:load, not running all the migrations\n# from scratch. The latter is a flawed and unsustainable approach (the more migrations\n# you'll amass, the slower it'll run and the greater likelihood for issues).\n#\n# It's strongly recommended that you check this file into your version control system.\n\nActiveRecord::Schema.define(version: 0) do\n\n  create_table \"virtual_columns\", force: :cascade, options: \"ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci\" do |t|\n    t.string \"name\"\n    t.virtual \"upper_name\", type: :string, as: \n    t.virtual \"name_length\", type: :integer, as: , stored: true\n  end\n\nend\n".


bin/test test/cases/adapters/mysql2/virtual_column_test.rb:53



Finished in 0.476529s, 2.0985 runs/s, 4.1970 assertions/s.

1 runs, 2 assertions, 1 failures, 0 errors, 0 skips
$
```

### System configuration
**Rails version**: master branch

**Ruby version**:ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux-gnu]

**Comment by yahonda** [comment]
It may be related with #29670.  Rails virtual column support for MySQL and MariaDB implemented at #22589 . Unit tests are executed with MariaDB 10.0 using Travis CI.

There are some changes at MariaDB 10.2. 
https://mariadb.com/kb/en/mariadb/virtual-computed-columns/

**Comment by yahonda** [comment]
These two tests are running fine with MariaDB 10.1.25.

```sql
Server version: 10.1.25-MariaDB-1~zesty mariadb.org binary distribution
```
